port/unix/omrsysinfo.c

```diff
@@ -6495,140 +6490,159 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
  *
- * @return 0 on success, otherwise negative error code
+ * @return TRUE if running in a container; otherwise, return FALSE
  */
-static int32_t
-isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
+static BOOLEAN
+isRunningInContainer(struct OMRPortLibrary *portLibrary)
 {
-	int32_t rc = 0;
-	FILE *cgroupFile = NULL;
+	if (OMRPORT_RUNNING_IN_CONTAINER_UNINITIALIZED == PPG_isRunningInContainer) {
+		FILE *cgroupFile = NULL;
+		int32_t rc = 0;
 
-	/* Assume we are not in container */
-	*inContainer = FALSE;
+		/* Assume, we are not in a container. */
+		BOOLEAN inContainer = FALSE;
 
-	/* Check for existence of files that signify running in a container (Docker and Podman respectively). Not completely
-	 * reliable as these files may not be available, but it should work for most cases.
-	 */
-	if ((0 == access("/.dockerenv", F_OK)) || (0 == access("/run/.containerenv", F_OK))) {
-		*inContainer = TRUE;
-	} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
-		/* Read PID 1's cgroup file /proc/1/cgroup and check cgroup name for each subsystem.
-		 * If cgroup name for each subsystem points to the root cgroup "/",
-		 * then the process is not running in a container.
-		 * For any other cgroup name, assume we are in a container.
-		 * Note that this will not work if namespaces are enabled in the container as all
-		 * cgroup names will be the root cgroup.
+		/* Check for existence of files that signify running in a container (Docker and Podman respectively). Not completely
+		 * reliable as these files may not be available, but it should work for most cases.
 		 */
-		cgroupFile = fopen(OMR_PROC_PID_ONE_CGROUP_FILE, "r");
-
-		if (NULL == cgroupFile) {
-			int32_t osErrCode = errno;
-			Trc_PRT_isRunningInContainer_fopen_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
-			rc = portLibrary->error_set_last_error(portLibrary, osErrCode, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED);
-			goto _end;
-		}
-
-		while (0 == feof(cgroupFile)) {
-			char buffer[PATH_MAX];
-			char cgroup[PATH_MAX];
-			char subsystems[PATH_MAX];
-			int32_t hierId = -1;
-
-			if (NULL == fgets(buffer, sizeof(buffer), cgroupFile)) {
-				break;
-			}
-			if (0 != ferror(cgroupFile)) {
+		if ((0 == access("/.dockerenv", F_OK)) || (0 == access("/run/.containerenv", F_OK))) {
+			inContainer = TRUE;
+		} else if (OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE)) {
+			/* Read PID 1's cgroup file /proc/1/cgroup and check cgroup name for each subsystem.
+			 * If cgroup name for each subsystem points to the root cgroup "/",
+			 * then the process is not running in a container.
+			 * For any other cgroup name, assume we are in a container.
+			 * Note that this will not work if namespaces are enabled in the container as all
+			 * cgroup names will be the root cgroup.
+			 */
+			if (NULL == (cgroupFile = fopen(OMR_PROC_PID_ONE_CGROUP_FILE, "r"))) {
```

Comment:
Please keep the assignment separate from the test:
```c
			cgroupFile = fopen(OMR_PROC_PID_ONE_CGROUP_FILE, "r");
			if (NULL == cgroupFile) {
```

Reply:
The assignments have been separated from the test.